### PR TITLE
[Feature] Add option to save octomap when octomap node exits

### PR DESCRIPTION
**Pull request by ChenyueLu**
Add option for octomap server node to save cached octomap into file when it exits.

### CODE REVIEW

File: src/octomap_server.cpp [hunk 1]
```diff
@@ -179,7 +181,16 @@ namespace octomap_server {
     }
 
     OctomapServer::~OctomapServer() {
-
+        RCLCPP_INFO(this->get_logger(), "\033[34mNode shutdown <<< OctomapServer\033[0m");
```
Comment (rcwchan) [inline src/octomap_server.cpp:184]: Do I understand correctly that this is for bash colouring? `\033[34m`

File: src/octomap_server.cpp [hunk 2]
```diff
@@ -179,7 +181,16 @@ namespace octomap_server {
...
+        // Save octomap if needed
+        if (m_saveWhenExit) {
+            RCLCPP_INFO(this->get_logger(), "Saving octomap before exit....");
+            bool succeed = saveMap();
+            if (succeed)
+                RCLCPP_INFO(this->get_logger(), "Octomap successfully been saved when exits");
+            else
+                RCLCPP_ERROR(this->get_logger(), "Save Octomap failed when exits!");
```
Comment (rcwchan) [inline src/octomap_server.cpp:192]: please change to

```
Octomap failed to save when exiting!
```

File: src/octomap_server.cpp [hunk 3]
```diff
@@ -179,7 +181,16 @@ namespace octomap_server {
...
 
     OctomapServer::~OctomapServer() {
-
+        RCLCPP_INFO(this->get_logger(), "\033[34mNode shutdown <<< OctomapServer\033[0m");
+        // Save octomap if needed
+        if (m_saveWhenExit) {
+            RCLCPP_INFO(this->get_logger(), "Saving octomap before exit....");
+            bool succeed = saveMap();
+            if (succeed)
+                RCLCPP_INFO(this->get_logger(), "Octomap successfully been saved when exits");
```
Comment (rcwchan) [inline src/octomap_server.cpp:190]: please change to

```
Octomap has been successfully saved when exiting!
```

**Review by rcwchan — APPROVED** [review]
Please update the logs!

Other than that, LGTM!

File: src/octomap_server.cpp (hunk 1 above)
Comment (ChenyueLu) [reply src/octomap_server.cpp:184]: Yes!!!

File: src/octomap_server.cpp (hunk 3 above)
Comment (ChenyueLu) [reply src/octomap_server.cpp:190]: Sure English Master!!!

**Pull request merged**